docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -608,6 +608,38 @@ You can also nest several DTO :
 	
 Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
 
+If you use your data-transfer objects for multiple query but not ever the same parameters, you can use named arguments or variadic arguments, add ``WithNamedArguments`` to your Dto :
+
+.. code-block:: php
+
+    <?php
+
+    use Doctrine\ORM\WithNamedArguments;
+
+    class CustomerDTO implements WithNamedArguments
+    {
+        public function __construct(public string|null $name = null, public string|null $email = null, public string|null $city = null, public mixed|null $value = null)
+        {
+        }
+    }
+
+And then you can select the fields you want in order you want :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a');
+    $users = $query->getResult(); // array of CustomerDTO
+
+you can either aliases column :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW CustomerDTO(c.name, CONCAT(a.city, ' ' , a.zip) AS value) FROM Customer c JOIN c.address a');
```

Comment:
Did you consider a syntax closer to PHP, like `value: CONCAT(a.city, ' ', a.zip)`?

Reply:
I can think about it but since we are in a SQL/DQL query I stayed in this way

Reply:
It works, but I really hate this notation in an SQL query, so I left AS keywork working too.